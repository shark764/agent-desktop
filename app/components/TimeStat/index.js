/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

/**
 *
 * TimeStat
 *
 */

import React from 'react';
import moment from 'moment';

import PropTypes from 'prop-types';

import Radium from 'radium';

function TimeStat(props) {
  const ms = moment.duration(props.time);
  let value = moment.utc(ms.asMilliseconds()).format('HH:mm:ss');
  if (props.unit === 'millis') {
    if (value.slice(0, 5) === '00:00') {
      let seconds = value.slice(6);
      if (seconds.slice(0, 1) === '0') {
        seconds = seconds.slice(1);
      }
      value = `${seconds}s`;
    } else if (value.slice(0, 2) === '00') {
      value = value.slice(3);
    }
  } else {
    throw new Error('Not handling time not in millis');
  }

  return (
    <div style={props.style}>
      {value}
    </div>
  );
}

TimeStat.propTypes = {
  style: PropTypes.array,
  time: PropTypes.number.isRequired,
  unit: PropTypes.string.isRequired,
};

export default Radium(TimeStat);
