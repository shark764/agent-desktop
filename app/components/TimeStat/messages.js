/*
 * Copyright Â© 2015-2020 Serenova, LLC. All rights reserved.
 */

/*
 * TimeStat Messages
 *
 * This contains all the text for the TimeStat component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  secondsOneLetterAbbreviation: {
    id: 'app.components.TimeStat.secondsOneLetterAbbreviation',
    defaultMessage: 's',
  },
});
