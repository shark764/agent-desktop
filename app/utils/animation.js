/**
 * throttle max function call rate to browser refresh rate
 */

const requestAnimationFrame =
  window.requestAnimationFrame ||
  window.webkitRequestAnimationFrame ||
  window.mozRequestAnimationFrame ||
  ((callback) => window.setTimeout(callback, 1000 / 60));

function throttler(...callbackArgs) {
  if (!this.running) {
    this.running = true;
    requestAnimationFrame(() => {
      this.callback(...callbackArgs);
      this.running = false;
    });
  }
}

export function throttleDecorator(funcToThrottle) {
  return throttler.bind({ callback: funcToThrottle, running: false });
}
