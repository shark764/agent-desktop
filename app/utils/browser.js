/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

// Check if browser used is IE11 or greater
// Returns undefined for Chrome|Firefox|Edge
export const isIeEleven = () => document.documentMode && document.documentMode >= 11;
