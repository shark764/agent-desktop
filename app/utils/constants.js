/*
 * Copyright Â© 2015-2020 Serenova, LLC. All rights reserved.
 */

export const colors = {
  lightGray: '#ebebeb',
  offWhite: '#e4e4e4',
  lightRed: '#FE4565',
  regularRed: '#E43D5A',
  darkRed: '#CB3750',
  darkGreen: '#001b1e',
  hoverDarkGreen: 'rgba(0, 27, 30, 0.85)',
  black: '#000',
  white: '#FFFFFF',
  skylightBlue: '#23cdf4',
  toggleChecked: '#54B84F',
  toggleCheckedDisabled: '#b7e3b3',
};
