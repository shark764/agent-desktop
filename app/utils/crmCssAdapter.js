/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

/*
  * crmCssAdapter
  *
  * Updates CSS as needed to work with various CRM integrations
  *
  */

export default function crmCssAdapter (styleObject, styleNameArray, crmModule) {
  let styleAffix;
  if (crmModule === 'salesforce-lightning') {
    styleAffix = 'SfLightning';
  }

  styleNameArray.forEach((styleName) => {
    Object.assign(
      styleObject[styleName],
      styleObject[`${styleName}${styleAffix}`]
    );
  });
}