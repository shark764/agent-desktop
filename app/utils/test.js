import { IntlProvider } from 'react-intl';
import enUS from 'translations/en-US.json';

export const getIntlContext = () => {
  // Construct a new `IntlProvider` instance by passing `props` and
  // `context` as React would, then call `getChildContext()` to get the
  // React Intl API, complete with the `format*()` functions.
  const intlProvider = new IntlProvider({ locale: 'en-US', messages: enUS }, {});
  const { intl } = intlProvider.getChildContext();
  return intl;
};

export default { getIntlContext };
