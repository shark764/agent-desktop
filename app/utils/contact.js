/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import { isValidEmail, isPossibleNumber } from 'utils/validator';
import isURL from 'validator/lib/isURL';

export const formatValue = (attributeToValidate, value) => {
  let formattedValue;
  switch (attributeToValidate.type) {
    case 'phone':
      formattedValue = value.replace(/[^0-9+*#]/g, '');
      if (formattedValue.indexOf('+') !== 0 && formattedValue.length > 0) {
        formattedValue = `+${formattedValue}`;
      }
      break;
    case 'boolean':
      if (value === 'false' || value === '') formattedValue = false;
      else formattedValue = !!value;
      break;
    default:
      formattedValue = value;
  }
  return formattedValue;
};

export const getError = (attributeToValidate, value) => {
  let error = false;
  if (
    attributeToValidate.mandatory &&
    attributeToValidate.type !== 'boolean' &&
    (value === undefined || value.trim().length < 1)
  ) {
    error = 'errorRequired';
  } else if (value !== undefined && value.length) {
    switch (attributeToValidate.type) {
      case 'email':
        if (!isValidEmail(value)) {
          error = 'errorEmail';
        }
        break;
      case 'phone':
        try {
          if (!isPossibleNumber(value)) {
            error = 'errorPhone';
          }
        } catch (e) {
          error = 'errorPhone';
        }
        break;
      case 'link':
        if (
          !isURL(value, {
            protocols: ['http', 'https'],
            require_protocol: true,
          })
        ) {
          error = 'errorLink';
        }
        break;
      case 'number':
        if (Number.isNaN(Number(value))) {
          error = 'errorNumber';
        }
        break;
      default:
        break;
    }
  }
  return error;
};

export const getLocaleLabel = (attribute, locale) => {
  if (attribute.label[locale] !== undefined) {
    return attribute.label[locale];
  } else if (attribute.label.all !== undefined) {
    // Special case for 'All'. We have our own local translation for it we use.
    return attribute.label.all;
  } else if (attribute.objectName !== undefined) {
    // Fallback for attributes
    return attribute.objectName;
  } else {
    // Fallback for section headers
    return '';
  }
};

export function createSearchQuery(query, nextPage) {
  const encodedQuery = {};
  Object.keys(query).forEach(queryName => {
    const queryToEncode = query[queryName];
    encodedQuery[queryName] = encodeURIComponent(queryToEncode);
  });
  return {
    query: Object.assign(encodedQuery, {
      page: nextPage !== undefined ? nextPage : 1,
    }),
  };
}
