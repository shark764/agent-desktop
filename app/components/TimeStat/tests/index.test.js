/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';
import { getIntlContext } from 'utils/test';

import TimeStat from '../index';

describe('<TimeStat />', () => {
  describe('when given required props, intl, and unit', () => {
    it('should render correctly', () => {
      const rendered = shallow(
        <TimeStat.WrappedComponent
          id="mockTimeStatId"
          intl={getIntlContext()}
          time={10}
          unit="millis"
        />
      );
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('when given required props, intl, and unit. Time in hours, minutes and seconds.', () => {
    it('should render correctly', () => {
      const rendered = shallow(
        <TimeStat.WrappedComponent
          id="mockTimeStatId"
          intl={getIntlContext()}
          time={1593031011}
          unit="millis"
        />
      );
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('when given required props, intl, and unit. Time in hours and minutes.', () => {
    it('should render correctly', () => {
      const rendered = shallow(
        <TimeStat.WrappedComponent
          id="mockTimeStatId"
          intl={getIntlContext()}
          time={90061}
          unit="millis"
        />
      );
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('when passed a unit not millis', () => {
    it('should throw an error', () => {
      let error;
      try {
        shallow(
          <TimeStat.WrappedComponent
            id="mockTimeStatId"
            intl={getIntlContext()}
            time={1}
            unit="other"
          />
        );
      } catch (e) {
        error = e;
      }
      expect(error.message).toEqual('Not handling time not in millis');
    });
  });
});
