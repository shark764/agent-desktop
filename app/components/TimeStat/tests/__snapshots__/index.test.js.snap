// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimeStat /> when given required props, intl, and unit should render correctly 1`] = `
<div>
  0s
</div>
`;

exports[`<TimeStat /> when given required props, intl, and unit. Time in hours and minutes. should render correctly 1`] = `
<div>
  01:30
</div>
`;

exports[`<TimeStat /> when given required props, intl, and unit. Time in hours, minutes and seconds. should render correctly 1`] = `
<div>
  10:30:31
</div>
`;
